### PR TITLE
#30 - Search risks in a 1000m radius

### PR DESCRIPTION
**Pull request by viniciusgama**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "pretest": "node ./tests/test_setup.js",
-    "test": "NODE_ENV=test ./node_modules/jasmine-node/bin/jasmine-node tests/",
+    "test": "node ./tests/test_setup.js && NODE_ENV=test ./node_modules/jasmine-node/bin/jasmine-node tests/",
```
Comment (pedroaugustofr) [inline package.json]: Should run test_setup only for functional tests

File: src/services/risk/risks-around.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      $near: [longitude, latitute],
+      $maxDistance: radiusInKm / kilometersPerDegree
+    }
+  }).exec(function(err, places) {
+    if (err) {
+      return console.error(err);
+    }
+
+    console.log("Places found: " + places.length);
+    console.log(places);
```
Comment (pedroaugustofr) [inline src/services/risk/risks-around.js]: Delete console.log

File: src/services/risk/risks-around.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+var Place = require('../../place.model');
+var isNumeric = require('is-numeric');
+
+module.exports = function(request, response) {
+  if (Object.keys(request.params).length === 0) {
+    return response.send(400, 'Invalid params.');
+  }
+
```
Comment (pedroaugustofr) [inline src/services/risk/risks-around.js]: Should 

```
if (Object.keys(request.params).length === 0) {
   return response.send(400, 'Invalid params.');
}
```

be extract to a validation method

File: src/services/risk/risks-around.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  var radiusInKm = 1;
+  var kilometersPerDegree = 111.2;
+  var longitude = request.params.longitude;
+  var latitute = request.params.latitude;
+
+  if (!isNumeric(latitute) || !isNumeric(longitude)) {
+    return response.send(400, 'Invalid latitude or longitude.');
+  }
+
```
Comment (pedroaugustofr) [inline src/services/risk/risks-around.js]: Should 

```
if (!isNumeric(latitute) || !isNumeric(longitude)) {
  return response.send(400, 'Invalid latitude or longitude.');
}
```

be extract to a validation method

File: tests/risks-around_spec.js [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
-    .toss();
-
-frisby.create('Should fail gracefully when latitude and longitude does not exists')
-    .get('http://localhost:8080/risks-around?latitude=90&longitude=90')
-    .expectStatus(204)
-    .toss();
+frisby.create('should return no content if not in 1km radius')
+  .get(risksAroundUrl + '?latitude=90&longitude=90')
+  .expectStatus(204)
+  .toss();
```
Comment (pedroaugustofr) [inline tests/risks-around_spec.js]: should move this file to `tests/integration` folder

File: src/services/risk/risks-around.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    console.log("Places found: " + places.length);
+    console.log(places);
+
+    if (places.length === 0) {
+      return response.send(204);
+    }
+
+    return response.send(places);
+  });
+};
```
Comment (pedroaugustofr) [inline src/services/risk/risks-around.js]: need unit tests for this controller

**Comment by tayanefernandes** [comment]
Any update on this story?

**Pull request merged**